pylint/checkers/typecheck.py

```diff
@@ -49,19 +49,18 @@
     supports_membership_test,
     supports_setitem,
 )
-from pylint.constants import PY310_PLUS
 from pylint.interfaces import HIGH, INFERENCE
 from pylint.typing import MessageDefinitionTuple
 
 if TYPE_CHECKING:
     from pylint.lint import PyLinter
 
-CallableObjects = Union[
-    bases.BoundMethod,
-    bases.UnboundMethod,
-    nodes.FunctionDef,
-    nodes.Lambda,
-    nodes.ClassDef,
+CallableObjects = type[
```

Comment:
mypy really wanted `type` in a few cases, and I skimmed https://mypy.readthedocs.io/en/stable/common_issues.html#variables-vs-type-aliases, but I'd need to study it more to understand why only certain places needed this change.